clients/pkg/promtail/targets/journal/journaltarget.go

```diff
@@ -67,10 +68,13 @@ var defaultJournalEntryFunc = func(c sdjournal.JournalReaderConfig, cursor strin
 	} else {
 		journal, err = sdjournal.NewJournal()
 	}
-
 	if err != nil {
 		return nil, err
-	} else if err := journal.SeekCursor(cursor); err != nil {
+	}
+	defer journal.Close()
```

Comment:
This would call `Close()` a second time, if it was already closed on line 95. Would that cause any problems?

Reply:
I had a look at the repository and found different resource closing patterns.

Therefore I have updated the pull request to an anonymous defer function to get rid of the double close.